### PR TITLE
replace boost::shared_lock with std version

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: main/EventSystem.cpp [hunk 1]
```diff
@@ -612,7 +613,7 @@ void CEventSystem::GetCurrentMeasurementStates()
 	m_windgustValuesByID.clear();
 	m_zwaveAlarmValuesByID.clear();
 
-	boost::shared_lock<boost::shared_mutex> devicestatesMutexLock(m_devicestatesMutex);
+	std::shared_lock<boost::shared_mutex> devicestatesMutexLock(m_devicestatesMutex);
```
Comment (gizmocuz) [inline main/EventSystem.cpp:616]: Is it maybe an idea to also use std::shared_mutex ... if it needs to be shared, maybe std::mutex will work as well ?

Comment (neheb) [reply main/EventSystem.cpp:616]: std::shared_mutex is C++17

Comment (gizmocuz) [reply main/EventSystem.cpp:616]: True, but does it need to be a shared mutex or is a normal mutex sufficient ?

Comment (neheb) [reply main/EventSystem.cpp:616]: I don't know. I prefer not to edit the code.

**Comment by gizmocuz** [comment]
I think this is currently been done in

#4715

Needs more testing, but so far no issues detected

File: main/EventSystem.cpp (hunk 1 above)
Comment (keryell) [reply main/EventSystem.cpp:616]: Is there a serious reason for not switching to C++17 anyway?

**Pull request closed without merging**